opm/polymer/fullyimplicit/BlackoilPolymerModel_impl.hpp

```diff
@@ -821,6 +825,51 @@ namespace Opm {
         for (size_t i = 0; i < nface; ++i) {
             water_vel[i] = water_vel[i] / (b_faces[i] * phiavg[i] * internal_face_areas[i]);
         }
+
+        // for SHRATE keyword treatment
+        if (has_shrate_) {
+
+            // get the upwind water saturation
+            const Opm::PhaseUsage pu = fluid_.phaseUsage();
+            const ADB& sw = state.saturation[pu.phase_pos[ Water ]];
+            const ADB& sw_upwind_adb = upwind.select(sw);
+
+            std::vector<double> sw_upwind;
+            sw_upwind.resize(sw_upwind_adb.size());
+            std::copy(&(sw_upwind_adb.value()[0]), &(sw_upwind_adb.value()[0]) + sw_upwind_adb.size(), sw_upwind.begin());
+
+            // get the absolute permeability for the faces
+            std::vector<double> perm;
+            perm.resize(transi.size());
+
+            for (size_t i = 0; i < transi.size(); ++i) {
+                perm[i] = transi[i] / internal_faces[i];
+            }
+
+            // get the upwind krw_eff
+            const ADB& krw_adb = upwind.select(krw_eff);
+            std::vector<double> krw_upwind;
+            krw_upwind.resize(krw_adb.size());
+            std::copy(&(krw_adb.value()[0]), &(krw_adb.value()[0]) + krw_adb.size(), krw_upwind.begin());
```

Comment:
Use `x.data()` instead of `&x[0]`. This applies both to `Eigen::Array`s and `std::vector`s.